sslh + libconfig as a req

benf33 opened the pull request:
Adding new component
sslh and libconfig as a requirement for sslh

Commit:
Add component libconfig as a requirement for sslh

Commit:
Add component sslh

benf33 commented:
- sslh.8.gz is not in gzip format (make publish scramble the gzip format of the file)
- service doesn't start with SMF while ok with command line (sslh fails to setid/gid to nobody)

Commit:
Consistency between hardcoded configuration file lookup and method

Commit:
Remove unused commented properties

Commit:
No need to specify SMG FMRI, defined by XML file

pyhalov commented:
We don't deliver gzipped manual pages. They shouldn't be compressed

pyhalov commented on components/libconfig/Makefile:
```diff
@@ -0,0 +1,51 @@
+#
```
Please, use new header from transforms/copyright-template

benf33 replied:
done

pyhalov commented on components/libconfig/libconfig.p5m:
```diff
@@ -0,0 +1,60 @@
+restart_fmri=svc:/application/texinfo-update:default facet.doc.info=true 
+file path=/usr/include/libconfig.h
+file path=usr/include/libconfig.h++
+file path=usr/lib/libconfig.la
```
don't include *la, *a files

pyhalov replied:
You can use gmake sample-manifest to generate p5m file

benf33 replied:
When I do this, it generates a new file with the following header :

# A full copy of the text of the CDDL should have accompanied this

# source. A copy of the CDDL is also available via the Internet at

# http://www.illumos.org/license/CDDL.

Like the header says, should I include the 377 lines license from the website or leave the text that suggests that something have been done improperly ?

Best regards.
Ben

----- Mail original -----
De: "Alexander Pyhalov" notifications@github.com
À: "OpenIndiana/oi-userland" oi-userland@noreply.github.com
Cc: "benf33" bentahyr@chez.com
Envoyé: Mardi 16 Février 2016 20:31:32
Objet: Re: [oi-userland] sslh + libconfig as a req (#1701)

In components/libconfig/libconfig.p5m : > +set name=pkg.summary value="C/C++ library for processing configuration files"

> +set name=pkg.description \
> -    value="C/C++ library for processing configuration files"
>   +set name=com.oracle.info.description value="C/C++ library for processing configuration files"
>   +set name=info.classification \
> -    value=org.opensolaris.category.2008:System/Libraries
>   +set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
>   +set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
>   +set name=org.opensolaris.arc-caseid value=PSARC/2009/118 value=PSARC/2013/044
>   +set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
>   +dir path=usr/share/info
>   +file path=usr/share/info/libconfig.info group=bin mode=0444 owner=root \
>   +restart_fmri=svc:/application/texinfo-update:default facet.doc.info=true 
>   +file path=/usr/include/libconfig.h
>   +file path=usr/include/libconfig.h++
>   +file path=usr/lib/libconfig.la 

You can use gmake sample-manifest to generate p5m file 

— 
Reply to this email directly or view it on GitHub .

pyhalov commented on components/sslh/sslh.p5m:
```diff
@@ -0,0 +1,44 @@
+file path=usr/sbin/sslh
+file path=usr/share/man/man8/sslh.8.gz
+legacy pkg=SUNWsslh desc="Applicative protocol multiplexer" \
+    name="sslh - Applicative protocol multiplexer"
```
No need to deliver legacy action

benf33 replied:
Done

pyhalov commented on components/sslh/sslh.p5m:
```diff
@@ -0,0 +1,44 @@
+    value="org.opensolaris.category.2008:System/Security"
+set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
+set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=org.opensolaris.arc-caseid \
```
Where did you take arc-caseid? We don't use them.

benf33 replied:
took it from ntp component

pyhalov replied:
I'd regenerate manifest

pyhalov commented:
As libconfig is sslh requirement, it should be added by separate commit before sslh.

pyhalov commented on components/libconfig/Makefile:
```diff
@@ -0,0 +1,51 @@
+COMPONENT_PRE_CONFIGURE_ACTION =        (rm $(SOURCE_DIR)/lib/scanner.{c,h} && $(CLONEY) $(SOURCE_DIR) $(@D) )
+
+# common targets
+build:		$(BUILD_32)
```
Ideally, library should deliver both 32 and 64 bit versions

benf33 replied:
I'll redo a specific pull request for libconfig in both 32/64 bits.

Commit:
Fixed P5M file :
- build from make sample-manifest
- remove .a .la files
- include 64bits files as well

Fixed Makefile :
- include both 32 and 64 bits during build and install

pyhalov commented on components/libconfig/libconfig.p5m:
```diff
@@ -0,0 +1,58 @@
+#
+
+#
+# Copyright 2016 N/A
```
Please, insert your firstname and lastname

benf33 replied:
I would prefer not to have my name on anything on the Internet. I'll put any name you happy with, by default John Smith.

pyhalov commented on components/libconfig/libconfig.p5m:
```diff
@@ -0,0 +1,58 @@
+
+file path=usr/include/libconfig.h
+file path=usr/include/libconfig.h++
+dir  path=usr/lib/cmake
```
remove dir actions

benf33 replied:
All of them or just the first one ?
Should I assume IPS will make the directory if it doesn't exist ?

pyhalov commented on components/libconfig/libconfig.p5m:
```diff
@@ -0,0 +1,58 @@
+dir  path=usr/share/info
+dir  path=usr/share/info/lib
+file path=usr/share/info/libconfig.info group=bin mode=0444 owner=root \
+restart_fmri=svc:/application/texinfo-update:default facet.doc.info=true 
```
insert several spaces or tab

benf33 replied:
Done

pyhalov commented on components/sslh/Makefile:
```diff
@@ -0,0 +1,65 @@
+# remove the unwanted file.
+COMPONENT_PREP_ACTION = rm -f $(SOURCE_DIR)/match.S
+
+COMPONENT_BUILD_ARGS += -f Makefile
```
Is it really necessary?

benf33 replied:
Stripped out all unnecessary stuff

pyhalov commented on components/sslh/Makefile:
```diff
@@ -0,0 +1,65 @@
+COMPONENT_PREP_ACTION = rm -f $(SOURCE_DIR)/match.S
+
+COMPONENT_BUILD_ARGS += -f Makefile
+COMPONENT_BUILD_TARGETS += CC=gcc
```
I don't think BUILD_TARGETS is a good way to set CC. Why also CC=gcc and not $(CC)? I'd set it in COMPONENT_BUILD_ENV

benf33 replied:
If I dont put COMPONENT_BUILD_TARGETS += CC=gcc, it picks up cc as compiler rather than gcc

pyhalov commented on components/sslh/Makefile:
```diff
@@ -0,0 +1,65 @@
+
+COMPONENT_BUILD_ARGS += -f Makefile
+COMPONENT_BUILD_TARGETS += CC=gcc
+COMPONENT_INSTALL_ARGS += -f Makefile
```
Why do you explicitly specify Makefile name?

benf33 replied:
Stripped out these entries

Commit:
Removed dir actions
Added tabs to aligned action parameters

Commit:
Change license header from Opensolaris to Illumos CDDL

Commit:
Update license headers from Opensolaris to Illumos CDDL
Clean Makefile of unnecessary bits
Clean p5m of unnecessary bits
p5m generated from Makefile info
Patch to sslh Makefile includes not compressed man page
SMF manifest has been stripped out of unnecessary things
SMF method script has been completely rewritten

pyhalov commented on components/libconfig/libconfig.p5m:
```diff
@@ -0,0 +1,48 @@
+file path=usr/lib/libconfig.so.9.2.0
+file path=usr/lib/pkgconfig/libconfig++.pc
+file path=usr/lib/pkgconfig/libconfig.pc
+file path=usr/lib/amd64/cmake/libconfig++/libconfig++Config.cmake
```
Please, use $(MACH64) instead of amd64 to be more SPARC-friendly

pyhalov commented:
Also, please create separate pull requests for sslh and liibconfig. Move components to corresponding directories according to http://wiki.openindiana.org/oi/Categories+layout

Commit:
Use of  in p5m file

pyhalov commented on components/sslh/patches/00-makefile.patch:
```diff
@@ -0,0 +1,23 @@
+ 
+ $(MAN): sslh.pod Makefile
+-	pod2man --section=8 --release=$(VERSION) --center=" " sslh.pod | gzip -9 - > $(MAN)
++	/usr/perl5/5.22/bin/pod2man --section=8 --release=$(VERSION) --center=" " sslh.pod > $(MAN)
```
Don't do this, this will be broken when perl is updated. Set PATH in Makefile to /usr/gnu/bin:/usr/bin:/usr/perl5/bin
